packages/botonic-react/src/components/multichannel/multichannel-text.jsx

```diff
@@ -77,11 +81,14 @@ export const MultichannelText = props => {
       const asText =
         type === buttonTypes.POSTBACK ? postbackButtonsAsText : true
       const newline =
-        multichannelContext.messageSeparator == null &&
+        multichannelContext.messageSeparator === null &&
         !newLineFirstButton &&
         i === 0
           ? ''
           : '\n'
+      if (type !== buttonTypes.POSTBACK && multichannelButton.props.payload) {
+        delete multichannelButton.props.payload
+      }
```

Comment:
Doubt: When will this happen? Under what circumstances can we have a non POSTBACK button with a payload?

Reply:
Now with flow builder we are adding in all buttons a part at the end of the payload. To make sure that all payloads of a text or whatsapp button list are different. What happens is that it is always added if the payload is empty it is also added, maybe we should change this logic.

Maybe we should change this logic so that if it is a button with url but without payload is not added this final part of the payload.

```typescript
return (
      <Button
        key={this.id}
        url={this.url}
        payload={this.getButtonPayload(buttonIndex)}
      >
        {this.text}
      </Button>
    )
  }

  private getButtonPayload(buttonIndex: number): string {
    return `${this.payload}${SOURCE_INFO_SEPARATOR}${buttonIndex}`
  }
```

Reply:
I have modified this logic in the flow builder plugin

Reply:
Yes, we have to adapt this logic in the Flow Builder's plugin. The code of a Botonic plugin should never affect the code of Botonic, the plugins have to be the ones to adapt to Botonic's code.